tests/e2e/specs/stories-editor/text-block.test.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { createNewPost, dragAndResize, getEditedPostContent } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { activateExperience, deactivateExperience, selectBlockByClassName } from '../../utils';
+
+describe( 'Text Block', () => {
+	beforeAll( async () => {
+		await activateExperience( 'stories' );
+	} );
+
+	afterAll( async () => {
+		await deactivateExperience( 'stories' );
+	} );
+
+	const textBlockClass = 'wp-block-amp-story-text';
+
+	beforeEach( async () => {
+		await createNewPost( { postType: 'amp_story' } );
+		await page.waitForSelector( `.${ textBlockClass }.is-not-editing` );
+		await selectBlockByClassName( textBlockClass );
+	} );
+
+	it( 'should not be in editable mode when selecting the block', async () => {
+		await page.keyboard.type( 'Hello' );
+		const content = await page.$eval( '.block-editor-block-list__layout .block-editor-block-list__block .wp-block-amp-amp-story-text', ( node ) => node.textContent );
+		expect( content ).toStrictEqual( 'Write text…' );
+	} );
+
+	it( 'should go to editable mode after two clicks', async () => {
+		const textToWrite = 'Hello';
+
+		await page.click( `.${ textBlockClass }` );
+		await page.keyboard.type( textToWrite );
+		const content = await page.$eval( '.block-editor-block-list__layout .block-editor-block-list__block .wp-block-amp-amp-story-text', ( node ) => node.textContent );
+		expect( content ).toStrictEqual( textToWrite );
+	} );
+
+	// @todo Dragging is not working for some reason.
+	it.skip( 'should allow dragging the Text block from anywhere when not editing', async () => { // eslint-disable-line jest/no-disabled-tests
+		const textBlock = await page.$( '.wp-block-amp-story-text-wrapper' );
+		await dragAndResize( textBlock, { x: 0, y: 25 } );
```

Comment:
Can't get the dragging to work, out of ideas for this moment. Perhaps you have any ideas, @swissspidy?
If not then maybe we could create a separate issue for investigating dragging + e2e.

Reply:
Have never tested dragging with Puppeteer before, but I can have a look.

Reply:
There's an overlay over the page block by default which might cause problems (`.block-editor-inner-blocks.has-overlay::after).

In my testing the text block move once I interacted with the page while debugging the test. So _something_ happens.

Reply:
I also saw something happening but what made it hard to debug was that sometimes nothing happened and sometimes something happened. For example, once a new tab opened with the block attributes displayed in the search bar (e.g. `{"type":"block","srcIndex":0,"srcRootClientId":"","srcClientId":"6d9d675d-768c-4e27-85a9-d3520739e243"}` 🤷‍♀

Reply:
> There's an overlay over the page block by default which might cause problems (`.block-editor-inner-blocks.has-overlay::after).

In theory, clicking on the block to make sure that the block is selected first should get rid of the overlay. But perhaps it's not happening as it should.

It could be worthwhile trying if the dragging works on a regular post, although that might be not easy to do / debug since the dropzone is quite narrow.

Reply:
> For example, once a new tab opened with the block attributes displayed in the search bar

Had that too once or twice. Maybe an issue in the upstream e2e utils?

---

I'll try to look at it again tonight to see if I can find out more.

Reply:
No luck so far. Will indeed have to try with something else for starters (reorder panel, regular block mover in the block editor, etc.).

Maybe I am missing something, but `dragAndResize` should work in theory (despite the name)

Reply:
It should work for both dragging and resizing (since that's also dragging basically). Not sure why it's not working. It's used for e2e tests upstream as well, however, not 100% sure if it actually works there either since that test is just using the Snapshots and snapshots assume that the test works OK in the first place.

Maybe we should create a separate issue for figuring out the dragging? This will be necessary for other e2e tests as well, such as the snap + rotation; resizing, reordering, etc.

Thoughts?

Reply:
The upstream snapshots looked OK though when I checked these today.

Will give it one more try, then we can create a new issue.

Reply:
Just checked that the shapshot does seem to look OK indeed.

Reply:
Opened a new issue to explore this: #2975.